src/d3d9/d3d9_device.cpp

```diff
@@ -1445,12 +1445,17 @@ namespace dxvk {
     }
 
     auto ClearImageView = [this](
-      bool                     fullClear,
+      uint32_t                 alignment,
       VkOffset3D               offset,
       VkExtent3D               extent,
       const Rc<DxvkImageView>& imageView,
       VkImageAspectFlags       aspectMask,
       VkClearValue             clearValue) {
+      
+      VkExtent3D imageExtent = imageView->image()->info().extent;
+      const bool fullClear = align(extent.width, alignment) == imageExtent.width
```

Comment:
I would clamp the extent before this, then we can still do fast clears if the extent is too big.

Reply:
Also you need to check if the offset is 0

Reply:
Good call.